showmehow/showmehow.py

```diff
@@ -338,13 +338,14 @@ def handle_attempt_lesson_remote(self, source, result):
         for side_effect in side_effects:
             dispatch_side_effect(side_effect, self._coding_game_service)
 
+        # Regardless of what the lesson is, fire this event so that
+        # the game service can know that *a* lesson completed.
+        dispatch_side_effect({
+            "type": "event",
+            "value": "showmehow-lesson-completed"
```

Comment:
In this case, the event should not be `lesson-completed`, since the lesson is not completed. You could change it to `task-completed`, or add another one for that case.